src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -2757,6 +2757,86 @@ static QVariant fcnGeometryCollectionAsArray( const QVariantList &values, const
   return array;
 }
 
+static QVariant fcnGeometryDissolve( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QgsGeometry geom;
+  QVector<QgsGeometry> geometries;
+  for ( int i = 0; i < values.count(); i++ )
+  {
+    geom = QgsExpressionUtils::getGeometry( values.at( i ), parent );
+    if ( geom.isNull() )
+      continue;
+    geometries.append( geom.asGeometryCollection() );
+  }
+  QgsGeometry dissolved = QgsGeometry().unaryUnion( geometries );
+  return QVariant::fromValue( dissolved );
+}
+
+static QVariant fcnSplitGeometry( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QgsGeometry initGeom = QgsExpressionUtils::getGeometry( values.at( 0 ), parent );
+  QgsGeometry inGeom = initGeom;
+  if ( inGeom.isNull() || inGeom.type() == QgsWkbTypes::PointGeometry )
+    return QVariant::fromValue( initGeom );
+  QgsGeometry splitGeom = QgsExpressionUtils::getGeometry( values.at( 1 ), parent );
+  if ( splitGeom.isNull() || splitGeom.type() == QgsWkbTypes::PointGeometry )
+    return QVariant::fromValue( initGeom );
+
+  QVector< QgsGeometry > outGeoms = initGeom.asGeometryCollection();
+  auto splitterFcn = [ = ]( const QVector< QgsGeometry > &sourceGeom, const QVector<QgsPoint> &line ) -> QVector< QgsGeometry >
+  {
+
+    QVector< QgsGeometry > outGeom;
+    QVector< QgsGeometry > newGeometries;
+    QgsPointSequence topologyTestPoints;
+
+    for ( const QgsGeometry &part : sourceGeom )
+    {
+      QgsGeometry partCopy = QgsGeometry( part );
+      QgsGeos geos( partCopy.get() );
+      QString error;
+      QgsGeometryEngine::EngineOperationResult result = geos.splitGeometry( line, newGeometries, true, topologyTestPoints, &error, false );
+
+      if ( result == QgsGeometryEngine::Success )
+      {
+        outGeom.append( newGeometries );
+      }
+      else
+      {
+        outGeom.append( partCopy );
+      }
+    }
+    return outGeom;
+  };
+
+  for ( auto splitPart = splitGeom.const_parts_begin(); splitPart != splitGeom.const_parts_end(); ++splitPart )
+  {
+    QgsGeometry partGeom( ( *splitPart )->clone() );
+    QVector<QgsPoint> splitter;
+    for ( auto pointIt = partGeom.vertices_begin(); pointIt != partGeom.vertices_end(); ++pointIt )
+      splitter.append( ( *pointIt ) );
```

Comment:
you could iterate over the vertices in splitPart directly here, instead of incurring the cost of the clone

Reply:
I recall that I tried something, but this was the only way I got it to work, I'll try to look into it again.